File: webconsole.client/client/app/domain/importPackage.ts
import {PackageResolvingCandidate} from '../domain/packageResolvingCandidate';

export interface ImportPackage {
    pkgName: string;
    version: string;
    resolution: string;
    versionRange: string;
    bundleSymbolicName: string;
    bundleVersion: string;
    packageResolvingCandidates: PackageResolvingCandidate[];
}

File: webconsole.osgi/src/io/skysail/webconsole/osgi/entities/packages/ImportPackage.java
package io.skysail.webconsole.osgi.entities.packages;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.osgi.service.packageadmin.ExportedPackage;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class ImportPackage extends PackageDescriptor {

    private Resolution resolution = Resolution.MANDATORY;
    private String versionRange;
    private String bundleSymbolicName;
    private String bundleVersion;

    private List<PackageResolvingCandidate> packageResolvingCandidates = new ArrayList<>();

    private Map<String, String> additionalAttributes = new HashMap<>();

    public ImportPackage(String pkgName) {
        super(pkgName);
    }

    public void addCandidate(ExportedPackage export) {
        packageResolvingCandidates.add(new PackageResolvingCandidate(export));
    }

    public void addAdditionalAttributes(String key, String value) {
        this.additionalAttributes.put(key, value);
    }

}
